docs/src/main/sphinx/security/jwt-auth.rst

```diff
@@ -0,0 +1,145 @@
+==================
+JWT authentication
+==================
+
+Trino can be configured to authenticate client access using `JSON web tokens
+<https://en.wikipedia.org/wiki/JSON_Web_Token>`_. A JWT is a small, web-safe
+JSON representation that contains the same cryptographic information as a
+certificate, including:
+
+-  Subject
+-  Valid time period
+-  Signature
+
+A `JWT <jwt.io>`_ is designed to be passed between servers as proof of prior
+authentication. An end user logs into a client application and requests access
+to a server. The server sends the user's credentials to a separate
+authentication service that validates the user, generates a JWT as proof of
+validation, then returns the JWT to the requesting server. The same JWT can then
+be forwarded to other services to maintain the user's validation without further
+credentials.
+
+.. important::
+
+    Do not configure JWT authentication in the belief that it is a part of
+    OAuth2 authentication.
+
+    Trino provides JWT authentication to support administrators at large sites
+    who are writing their own single sign-on or proxy system to stand between
+    users and the Trino coordinator, where their new system submits queries on
+    behalf of users.
+
+    JWT authentication is **not** expected to be needed or used by individuals
+    trying to connect to a Trino server.
+
+    Further, if another Trino-supported authentication system uses JSON web
+    tokens internally, such as :doc:`OAuth2 </security/oauth2>`, customers do
+    **not** need to configure JWT authentication as described on this page in
+    order to use that other authentication system. Use only the configuration
+    described on the other system's pages.
+
+Using JWT authentication
+------------------------
+
+A JWT can be passed between servers either in text form or encrypted with a
+standard encryption method described in `RFC 7519
+<https://datatracker.ietf.org/doc/html/rfc7519>`_.
+
+There are two ways to validate a JWT:
+
+-  Acquire its public key from a JSON web key set (JWKS) endpoint service (the
+   typical case)
+-  Extract the signature from the token and validate using a public key
+
+A JWKS endpoint is a read-only service that contains public key information in
+`JWK <https://datatracker.ietf.org/doc/html/rfc7517>`_ format. These public
+keys are the counterpart of the private keys embedded in JSON web tokens.
+
+JWT authentication configuration
+--------------------------------
+
+Do not configure JWT authentication unless you are an administrator trying to
+secure Trino with a custom single sign-on or proxy solution. If that describes
+you, then enable JWT authentication by setting the :doc:`JWT authentication type
+<authentication-types>` in ``etc/config.properties``:
+
+.. code-block:: properties
+
+    http-server.authentication.type=JWT
+
+In addition, you must configure the ``http-server.authentication.jwt.key-file``
```

Comment:
Since it is in the same file ... just collapse this into one codeblock

also link to the config.properities section in deployment.rst (ref is config_properties)

Reply:
Linked, but how to use ``...`` around a :ref: link?